client/app/components/Input/InputTag.jsx

```diff
@@ -15,185 +16,158 @@ export type Props = {
-          const { onCheckboxChange } = this.props;
-          if (onCheckboxChange) {
-            onCheckboxChange(newCheckbox);
-          }
+export function InputTag({
+  id,
+  name,
+  placeholder,
+  checkboxes: defaultCheckboxes,
+  onChange,
+  onCheckboxChange,
+}: Props) {
+  const [checkboxes, setCheckboxes] = useState<Checkbox[]>(defaultCheckboxes);
+  const [suggestions, setSuggestions] = useState<Checkbox[]>(defaultCheckboxes);
+  const [autocompleteLabel, setAutocompleteLabel] = useState<
+    string | void | null,
+  >(undefined);
+
+  const check = (inputId: string, checked: boolean) => {
+    const newCheckboxes = checkboxes.map((checkbox: Checkbox) => {
+      const newCheckbox = { ...checkbox };
+      if (newCheckbox.id === inputId) {
+        newCheckbox.checked = checked;
+        if (onCheckboxChange) {
+          onCheckboxChange(newCheckbox);
         }
-        return newCheckbox;
-      });
-      return checked
-        ? { checkboxes, autocompleteLabel: undefined }
-        : { checkboxes };
+      }
+      return newCheckbox;
     });
+
+    if (checked) {
+      setAutocompleteLabel(undefined);
+    }
+    setCheckboxes(newCheckboxes);
   };
 
-  checkboxChange = (checkbox: { checked: boolean, id: string }) => {
-    const { checked, id } = checkbox;
-    const { checkboxes } = this.state;
+  const checkboxChange = (checkbox: { checked: boolean, id: string }) => {
+    const { checked, id: inputId } = checkbox;
     if (
       !checked
-      && checkboxes.filter((item: Checkbox) => item.id === id && item.checked)
+      && checkboxes.filter((item: Checkbox) => item.id === inputId && item.checked)
         .length
     ) {
-      this.check(id, false);
+      check(inputId, false);
     }
   };
 
-  getSuggestions = (autocompleteLabel: string) => {
-    const inputValue = autocompleteLabel.trim().toLowerCase();
+  const getSuggestions = (label: string) => {
+    const inputValue = label.trim().toLowerCase();
     const inputLength = inputValue.length;
-    const { checkboxes } = this.state;
-    const suggestions: Checkbox[] = inputLength === 0
+    const newSuggestions: Checkbox[] = inputLength === 0
       ? checkboxes
       : checkboxes.filter(
         (checkbox: Checkbox) => checkbox.label.toLowerCase().indexOf(inputValue) > -1,
       );
-    return suggestions;
+    return newSuggestions;
   };
 
-  getSuggestionValue = (checkbox: Checkbox) => {
-    const { autocompleteLabel } = this.state;
-    return checkbox.label === autocompleteLabel ? checkbox.label : '';
-  };
+  const getSuggestionValue = (checkbox: Checkbox) => (checkbox.label === autocompleteLabel ? checkbox.label : '');
 
-  onSuggestionsFetchRequested = ({ value }: { value: string }) => {
-    const newSuggestions = this.getSuggestions(value);
-    this.setState({
-      suggestions: newSuggestions,
-    });
+  const onSuggestionsFetchRequested = (valueProp: { value: string }) => {
+    const { value } = valueProp;
```

Comment:
This had to be done this way because of https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unused-prop-types.md#false-positives-sfc.

Reply:
And @julianguyen, thanks for the pointer on the `react_component` error

Reply:
Ah thanks for explaining! Let's leave it in there for now then!